src/SDL_net.c

```diff
@@ -0,0 +1,1671 @@
+#include "SDL3_net/SDL_net.h"
+
+#ifdef __WINDOWS__
+// we need APIs that didn't arrive until Windows Vista.
+#ifdef _WIN32_WINNT
+  #if _WIN32_WINNT < 0x0600
+    #undef _WIN32_WINNT
+  #endif
+#endif
+#ifndef _WIN32_WINNT
+#define _WIN32_WINNT 0x0600
+#endif
```

Comment:
How about moving the block before including of `SDL_net.h`?
```suggestion
#ifdef _WIN32_WINNT
#  if _WIN32_WINNT < 0x0600 // we need APIs that didn't arrive until Windows Vista.
#    undef _WIN32_WINNT
#  endif
#  ifndef _WIN32_WINNT
#    define _WIN32_WINNT 0x0600
#  endif
#endif

#include "SDL3_net/SDL_net.h"

#ifdef __WINDOWS__
```

Reply:
Defining `_WIN32_WINNT` for non-windows won't hurt, I guess. If Ryan wants to avoid defining it through a build system, be that cmake or msvc project or whatever for whatever reason, this may be a solution.